python/ray/tests/test_placement_group.py

```diff
@@ -635,48 +634,161 @@ class NormalActor:
         placement_group_bundle_index=num_nodes * bundle_per_node - 1).remote()
-    # Destroy some nodes to fail placement group creation.
-    nodes_to_kill = get_other_nodes(cluster, exclude_head=True)
-    for node_to_kill in nodes_to_kill:
-        cluster.remove_node(node_to_kill)
 
     # Wait on the placement group now. It should be unready
     # because normal actor takes resources that are required
     # for one of bundle creation.
     ready, unready = ray.wait([pg.ready()], timeout=0)
     assert len(ready) == 0
     assert len(unready) == 1
+    # Wait until all tasks are done.
+    assert all(ray.get(tasks))
 
-    # Add a node back to schedule placement group.
-    for _ in range(len(nodes_to_kill)):
-        nodes.append(
-            cluster.add_node(num_cpus=bundle_cpu_size * bundle_per_node))
-    # Wait on the placement group creation.
+    # Wait on the placement group creation. Since resources are now available,
+    # it should be ready soon.
     ready, unready = ray.wait([pg.ready()])
     assert len(ready) == 1
     assert len(unready) == 0
 
     # Confirm that the placement group actor is created. It will
-    # raise an exception if actor was scheduled before placement group was
-    # created.
-    # TODO(sang): This with statement should be removed after atomic creation
-    # is implemented. It will be done in the next PR.
-    with pytest.raises(ray.exceptions.RayActorError):
-        ray.get(pg_actor.ping.remote(), timeout=3.0)
+    # raise an exception if actor was scheduled before placement
+    # group was created thus it checks atomicity.
+    ray.get(pg_actor.ping.remote(), timeout=3.0)
+    ray.kill(pg_actor)
+
+    # Make sure atomic creation failure didn't impact resources.
+    @ray.remote(num_cpus=bundle_cpu_size)
+    def resource_check():
+        return True
+
+    # This should hang because every resources
+    # are claimed by placement group.
+    check_without_pg = [
+        resource_check.remote() for _ in range(bundle_per_node * num_nodes)
+    ]
+
+    # This all should scheduled on each bundle.
+    check_with_pg = [
+        resource_check.options(
+            placement_group=pg, placement_group_bundle_index=i).remote()
+        for i in range(bundle_per_node * num_nodes)
+    ]
+
+    # Make sure these are hanging.
+    ready, unready = ray.wait(check_without_pg, timeout=0)
+    assert len(ready) == 0
+    assert len(unready) == bundle_per_node * num_nodes
+
+    # Make sure these are all scheduled.
+    assert all(ray.get(check_with_pg))
+
+    ray.util.remove_placement_group(pg)
+
+    def pg_removed():
+        return ray.util.placement_group_table(pg)["state"] == "REMOVED"
+
+    wait_for_condition(pg_removed)
+
+    # Make sure check without pgs are all
+    # scheduled properly because resources are cleaned up.
+    assert all(ray.get(check_without_pg))
+
+
+def test_mini_stress_scenario(ray_start_cluster):
+    # Create bundles as many as number of gpus in the cluster.
```

Comment:
This is not directly related to this PR, but I added it because we completed all P0 features.